dataclients/kubernetes/hosts.go

```diff
@@ -15,6 +15,9 @@ func createHostRx(hosts ...string) string {
 
 	hrx := make([]string, len(hosts))
 	for i, host := range hosts {
+		if strings.HasPrefix(host, "*.") {
+			host = strings.Replace(host, "*", "[a-z0-9]+((-[a-z0-9]+)?)*", 1)
```

Comment:
We need to verify this matches behaviour descibed at https://kubernetes.io/docs/concepts/services-networking/ingress/#hostname-wildcards

Makes sense to add this link as a comment.

Subdomain regexp looks strange with two capture groups using ? followed by *.

I guess subdomain shoud start and end with alphanumeric (not dash). Lets check this https://stackoverflow.com/questions/7930751/regexp-for-subdomain and similar answers and reuse/adopt regexp from there.

Reply:
Sounds like https://github.com/zalando/skipper/pull/3296#discussion_r1829017617 would be the one 😀

Reply:
I think this one already works for all cases.
![image](https://github.com/user-attachments/assets/c37357dc-2d9c-4c62-9a03-53342b846b67)

Reply:
> Sounds like #3296 (comment) would be the one

yes this also work